### PR TITLE
Add secret injection support for dictionary configurations

### PR DESCRIPTION
**Pull request by shishirx34**
Require this support for Synthetics.

### CODE REVIEW

**Comment by joelverhagen** [comment]
What do you think about making the naming/implementation mimic the "InMemory" provider documented here:
https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration-providers#memory-configuration-provider

This maybe will help understanding similar to how our JSON injected API mimics the existing non-injected JSON API.

**Comment by shishirx34** [comment]
@joelverhagen  - do you mean rename `Dictionary` to `InMemory`? Sounds better. will do.

**Comment by joelverhagen** [comment]
Yeah, essentially what I mean is try to make the API look exactly like the built-in in-memory one so it's more familiar to people with Microsoft.Extensions.Configuration experience.

File: src/NuGet.Services.Configuration/KeyVaultDictionaryInjectingConfigurationSource.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        public KeyVaultDictionaryInjectingConfigurationSource(IReadOnlyDictionary<string, string> dictionary, ISecretInjector secretInjector)
+        {
+            _dictionary = dictionary ?? throw new ArgumentNullException(nameof(dictionary));
+            _secretInjector = secretInjector ?? throw new ArgumentNullException(nameof(secretInjector));
+        }
+
+        public Extensions.IConfigurationProvider Build(IConfigurationBuilder builder)
+        {
+            var source = new DictionaryConfigurationSource(_dictionary);
```
Comment (agr) [inline src/NuGet.Services.Configuration/KeyVaultDictionaryInjectingConfigurationSource.cs]: Can we use [`MemoryConfigurationSource`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.memory.memoryconfigurationsource?view=dotnet-plat-ext-5.0) here?

**Review by agr — APPROVED** [review]

**Review by joelverhagen — APPROVED** [review]
Nice!

**Comment by lyndaidaii** [comment]
start reviewing while realized was merged.

**Comment by lyndaidaii** [comment]
LGTM 👍

**Pull request merged**